### PR TITLE
Include the target platform in the SBOM filename

### PR DESCRIPTION
**Pull request by Shnatsel**
And provide a `--no-target-suffix` flag to let people opt out.

### CODE REVIEW

File: cargo-cyclonedx/src/cli.rs [hunk 1]
```diff
@@ -63,6 +63,10 @@ Defaults to the host target, as printed by 'rustc -vV'"
     )]
     pub target: Option<String>,
 
+    /// Do not include the target platform of the BOM in the filename
+    #[clap(long = "no-target-suffix")]
+    pub no_target_suffix: bool,
```
Comment (lfrancke) [inline cargo-cyclonedx/src/cli.rs:68]: Personally I think this should be off by default.

File: cargo-cyclonedx/src/cli.rs [hunk 2]
```diff
@@ -154,21 +158,19 @@ impl Args {
...
-                prefix,
-            }),
-            (_, _) => None,
-        };
+        let output_options =
+            if cdx_extension.is_none() && prefix.is_none() && !self.no_target_suffix {
+                None
+            } else {
+                Some(OutputOptions {
+                    cdx_extension: cdx_extension.unwrap_or_default(),
```
Comment (lfrancke) [inline cargo-cyclonedx/src/cli.rs:166]: This comment doesn't concern this line in particular but in https://cyclonedx.org/specification/overview/#recognized-file-patterns it is outlined how a valid CycloneDX file is called and this PR violates that.

If it's not `bom.xml` anymore, it should be `bom_x86_64-unknown-linux-gnu.cdx.xml`

**Review by lfrancke — CHANGES_REQUESTED** [review]

**Comment by lfrancke** [comment]
To summarize my review:

- I see that people might want this, I don't think I'll need it but I'm not opposed
- I'd like to make this opt-in rather than opt-out
- We should stay compatible with the recognized file patterns which would imply to use the cdx suffix if target is included

**Comment by Shnatsel** [comment]
I believe I have addressed the review feedback, PTAL.

**Review by lfrancke — APPROVED** [review]

**Pull request merged**